code/modules/mob/living/carbon/human/life.dm

```diff
@@ -1782,19 +1782,12 @@
 /mob/living/carbon/human/handle_fire()
 	if(..())
 		return
-	var/thermal_protection = 0 //Simple check to estimate how protected we are against multiple temperatures
-	if(wear_suit)
-		if(wear_suit.max_heat_protection_temperature >= FIRESUIT_MAX_HEAT_PROTECTION_TEMPERATURE)
-			thermal_protection += (wear_suit.max_heat_protection_temperature*0.7)
-	if(head)
-		if(head.max_heat_protection_temperature >= FIRE_HELMET_MAX_HEAT_PROTECTION_TEMPERATURE)
-			thermal_protection += (head.max_heat_protection_temperature*THERMAL_PROTECTION_HEAD)
-	thermal_protection = round(thermal_protection)
-	if(thermal_protection >= FIRESUIT_MAX_HEAT_PROTECTION_TEMPERATURE)
-		bodytemperature += 11
-	else
-		bodytemperature += BODYTEMP_HEATING_MAX
-	return
+	
+	var/burn_temperature = fire_burn_temperature()
+	var/thermal_protection = get_heat_protection(burn_temperature)
+	
+	if (thermal_protection < 1 && bodytemperature < burn_temperature)
+		bodytemperature += round(BODYTEMP_HEATING_MAX*thermal_protection, 1)

```

Comment:
From what I can tell, this will result in maximum heating when the mob is almost but not quite fully protected, and no heating if unprotected. This is probably the reverse of the intended behaviour.